activerecord/lib/active_record/disable_joins_association_relation.rb

```diff
@@ -30,7 +30,9 @@ def load
         record[key]
       end
 
-      records = ids.flat_map { |id| records_by_id[id.to_i] }
+      records = ids.flat_map do |id|
+        records_by_id[id]
+      end
```

Comment:
I'd prefer this to be put back the way it was and just remove the `to_i`. When you make this change can you squash your commits into 1? Thanks!

Reply:
Yeah no worries — done 🙏